norse/torch/functional/stdp.py

```diff
@@ -1,60 +1,407 @@
-from typing import Callable, NamedTuple
+import torch
+
+from typing import Tuple, NamedTuple, Union
+
+from .heaviside import heaviside
+
+
+class STDPParameters(NamedTuple):
+    """Parameters of spike-timing-dependent plasticity (STDP)
+
+    Parameters:
+        a_pre (float): contribution of presynaptic spikes to trace
+        a_post (float): contribution of postsynaptic spikes to trace
+        tau_pre_inv (float): inverse decay time constant of presynaptic spike trace in 1/ms
+        tau_post_inv (float): inverse decay time constant of postsynaptic spike trace in 1/ms
+        w_min (float): lower bound on synaptic weight (should be < w_max)
+        w_max (float): upper bound on synaptic weight (should be > w_min)
+        eta_plus (float): contribution of pre-post activity to synaptic weight (should be >= 0)
+        eta_minus (float): contribution of post-pre activity to synaptic weight (should be >= 0)
+    """
+
+    a_pre: torch.Tensor = torch.as_tensor(1.0)
+    a_post: torch.Tensor = torch.as_tensor(1.0)
+    tau_pre_inv: torch.Tensor = torch.as_tensor(1.0 / 100e-3)
+    tau_post_inv: torch.Tensor = torch.as_tensor(1.0 / 100e-3)
+    w_min: float = 0.0
+    w_max: float = 1.0
+    eta_plus: float = 1.0
+    eta_minus: float = 1.0
+
+
+# TODO: can be merged with above, but redundant passing around of values
```

Comment:
I think splitting these up is a good idea to ensure disambiguation

Reply:
Ok!

Reply:
I think it might be possible to merge the two if the STDP computation below is carried out on tensors or alternatively the update rule is a torch.nn.Module that carries these parameters.